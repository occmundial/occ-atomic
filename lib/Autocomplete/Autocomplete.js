import React from 'react';
import PropTypes from 'prop-types';
import autobind from 'class-autobind';
import cloneDeep from 'lodash/cloneDeep';
import idx from 'idx';

import TextField from '../TextField';
import Droplist from '../Droplist';

/**
 * This component integrates the TextField and Droplist components.
 * The value of TextField filters the list in Droplist.
 */
class Autocomplete extends React.Component {
	constructor(props) {
		super(props);
		autobind(this);
		this.state = {
			value: idx(props, _ => _.textfieldProps.valueProp) || '',
			focus: false,
			labels: []
		};
	}

	componentWillMount() {
		let { onRef } = this.props;
		if (onRef) onRef(this);
	}

	componentWillReceiveProps(props) {
		if (
			idx(props, _ => _.textfieldProps.valueProp) !==
			idx(this.props, _ => _.textfieldProps.valueProp)
		) {
			this.setState({ value: props.textfieldProps.valueProp });
		}
	}

	componentWillUnmount() {
		let { onRef } = this.props;
		if (onRef) onRef(undefined);
	}

	onChange(value) {
		const { onChange } = this.props;
		this.setState({ value, focus: true });
		if (onChange) onChange(value);
	}

	onFocus() {
		const { onFocus } = this.props;
		this.setState({ focus: true });
		if (onFocus) onFocus();
	}

	onBlur() {
		const { onBlur } = this.props;
		this.setState({ focus: false });
		if (onBlur) onBlur();
	}

	onKeyUp(key) {
		const { onKeyUp } = this.props;
		if (onKeyUp) onKeyUp(key);
	}

	onClear() {
		const { onClear } = this.props;
		this.setState({ value: '' });
		if (onClear) onClear();
	}

	onMouseDown(item) {
		const { onMouseDown, onChange, droplistProps } = this.props;
		this.setState({
			value: item[droplistProps.itemTextKey],
			focus: false
		});
		if (onChange) onChange(item[droplistProps.itemTextKey]);
		if (onMouseDown) onMouseDown(item);
	}

	onEnter(item) {
		const { onEnter, onChange, droplistProps } = this.props;
		this.setState({
			value: item[droplistProps.itemTextKey],
			focus: false
		});
		if (onChange) onChange(item[droplistProps.itemTextKey]);
		if (onEnter) onEnter(item);
	}

	forceValue(value) {
		this.setState({ value });
	}

	setRef(input) {
		if (input) this.input = input;
	}

	render() {
		const { value, focus } = this.state;
		const {
			classes,
			id,
			textfieldProps,
			droplistProps,
			className,
			style,
			showInitialData
		} = this.props;
		const showDropList = showInitialData
			? focus && droplistProps.items
			: value && focus && droplistProps.items;
		return (
			<div
				className={`${classes.autoComplete}${className ? ` ${className}` : ''}`}
				id={id}
				style={style}
				ref={ref => {
					this.autocomplete = ref;
				}}
			>
				<TextField
					{...textfieldProps}
					onRef={this.setRef}
					onChange={this.onChange}
					onFocus={this.onFocus}
					onBlur={this.onBlur}
					onKeyUp={this.onKeyUp}
					onClear={this.onClear}
					valueProp={value}
				/>
				{showDropList && (
					<Droplist
						{...droplistProps}
						items={cloneDeep(droplistProps.items)}
						term={value}
						onMouseDown={this.onMouseDown}
						onEnter={this.onEnter}
						isOnFocus
						className={`${classes.droplist}${
							droplistProps.className ? ` ${droplistProps.className}` : ''
						}`}
					/>
				)}
			</div>
		);
	}
}

Autocomplete.propTypes = {
	classes: PropTypes.object,
	/** Object with the props for the TextField component. Check their descriptions in the TextField docs. */
	textfieldProps: PropTypes.shape({
		id: PropTypes.string,
		type: PropTypes.string,
		name: PropTypes.string,
		label: PropTypes.string,
		placeholder: PropTypes.string,
		selectOnFocus: PropTypes.bool,
		clear: PropTypes.bool,
		iconName: PropTypes.string,
		theme: PropTypes.string,
		valueProp: PropTypes.string,
		className: PropTypes.string,
		inputClassName: PropTypes.string,
		disabled: PropTypes.bool,
		testId: PropTypes.string,
		disableAutoComplete: PropTypes.bool
	}).isRequired,
	/** Object with the props for the Droplist component. Check their descriptions in the Droplist docs. */
	droplistProps: PropTypes.shape({
		items: PropTypes.array,
		itemIdKey: PropTypes.string,
		itemTextKey: PropTypes.string,
		itemTextRightKey: PropTypes.string,
		groups: PropTypes.bool,
		groupIdKey: PropTypes.string,
		groupNameKey: PropTypes.string,
		groupItemsKey: PropTypes.string,
		className: PropTypes.string,
		filter: PropTypes.bool,
		testId: PropTypes.string
	}).isRequired,
	/** Function to call on change (TextField). */
	onChange: PropTypes.func,
	/** Function to call on key up (TextField). */
	onKeyUp: PropTypes.func,
	/** Function to call on focus (TextField). */
	onFocus: PropTypes.func,
	/** Function to call on blur (TextField). */
	onBlur: PropTypes.func,
	/** Function to call on clear (TextField). */
	onClear: PropTypes.func,
	/** Function to call on mouse down (Droplist item). */
	onMouseDown: PropTypes.func,
	onRef: PropTypes.func,
	id: PropTypes.string,
	className: PropTypes.string,
	style: PropTypes.object,
	/**  Shows dropList if it has initial data. */
	showInitialData: PropTypes.bool
};

Autocomplete.defaultProps = {
	textfieldProps: {},
	droplistProps: {},
	showInitialData: false
};

export default Autocomplete;
