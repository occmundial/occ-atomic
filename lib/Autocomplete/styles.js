import idx from 'idx';
import { createUseStyles } from 'react-jss';

import spacing from '../subatomic/spacing';

export default createUseStyles({
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: props => idx(props, _ => _.textfieldProps.label) || idx(props, _ => _.textfieldProps.lockHeight) ? spacing.xLarge : 40,
        left: 0,
        width: '100%'
    }
});
