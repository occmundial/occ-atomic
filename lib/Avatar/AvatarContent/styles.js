import fonts from '../../tokens/fonts.json';
import colors from '../../tokens/colors.json';
import borderRadius from '../../tokens/borderRadius.json';
import legacyFont from '../../subatomic/fonts';

const { avatar, text, border } = colors;
const transition = 'all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s';

const font = fonts['body-regular-strong'];

export default {
	wrap: {
		position: 'relative',
		width: '100%',
		height: '100%'
	},
	cont: {
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		borderRadius: borderRadius['br-full'],
		overflow: 'hidden',
		background: avatar.bg.default,
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center'
	},
	capital: {
		display: 'inline-block',
		font: `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFont.body}`,
		color: text.indigo.primary,
		gridRowStart: '1',
		gridColumnStart: '1',
		textAlign: 'center'
	},
	disabled: {
		pointerEvents: 'none',
		background: avatar.bg.disabled,
		'& $camera': {
			opacity: 1
		}
	},
	disabledCapital: {
		color: text.corp.disabled
	  },
	clickable: {
		transition,
			'&:hover': {
				background: avatar.bg.hover
			},
			'&:active': {
				background: avatar.bg.active
			}
	  },
	editable: {
		'&:hover $camera': {
			opacity: 1
		},
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	},
	person: {
		gridRowStart: 1,
		gridColumnStart: 1,
		transition
	},
	  photo: {
		border: `1px solid ${border.default.subtle}`
	},
	hoverPhoto: {
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	  },
	camera: {
		gridRowStart: 1,
		gridColumnStart: 1,
		opacity: 0,
		transition
	},
	overlay: {
		borderRadius: '50%',
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center',
		transition,
		border: `1px solid transparent`
	  }
};
