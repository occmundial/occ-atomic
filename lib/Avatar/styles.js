import { createUseStyles } from 'react-jss';

export default createUseStyles({
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
});
