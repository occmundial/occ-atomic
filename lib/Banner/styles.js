import spacing from '../subatomic/spacing';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

export default {
  container: { padding: spacing.small, backgroundColor: colors.ink },
  textBanner: { display: 'inline-block' },
  textFlex: {
    textAlign: 'start',
    [`@media screen and (min-width:${grid.xs}px)`]: { textAlign: 'center' },
  },
  iconFlex: { padding: spacing.small,
     [`@media screen and (min-width:${grid.xs}px)`]: { padding: 0 } },
  icon: { cursor: 'pointer' },
};
