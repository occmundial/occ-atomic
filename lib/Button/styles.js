import legacyColors from '../subatomic/colors';
import legacyFont from '../subatomic/fonts';
import icons from '../subatomic/icons';

import colors from '../tokens/colors.json';
import spacing from '../tokens/spacing.json';
import borderRadius from '../tokens/borderRadius.json';
import fonts from '../tokens/fonts.json';
import shadows from '../tokens/shadows.json';
import { legacyIcons } from '../subatomic/mappedIcons';

const { sec, secDark, grey500, secLight } = legacyColors;

const sm = fonts['button-small'];
const md = fonts['button-medium'];
const lg = fonts['button-large'];

const renderIcon = (props, color) => {
    if (!legacyIcons[props.icon]) return '';
    if (props.icon) {
        return icons.base(icons[props.icon].icon(color));
    } else if (props.iconRight) {
        return icons.base(icons[props.iconRight].icon(color));
    }
    return '';
};

export default {
    btn: {
        display: 'inline-block',
        boxSizing: 'border-box',
        position: 'relative',
        maxWidth: '100%',
        marginBottom: 0,
        border: 0,
        padding: [spacing['size-3'], spacing['size-4']],
        borderRadius: borderRadius['br-xs'],
        font: `${sm.fontWeight} ${sm.fontSize}/${sm.lineHeight} ${legacyFont.body}`,
        textAlign: 'center',
        textDecoration: 'none',
        whiteSpace: 'nowrap',
        transition: '0.3s all',
        cursor: 'pointer',
        userSelect: 'none',
        touchAction: 'manipulation',
        '&:focus': {
            outline: 'none'
        },
        '&:hover': {
            textDecoration: 'none'
        }
    },
    cont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        pointerEvents: 'none'
    },
    loadCont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'absolute',
        top: 0,
        left: 0,
        pointerEvents: 'none'
    },
    // Themes
    primary: {
        background: colors.button.primary.bg.default,
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.primary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.primary.bg.hover
        },
        '&:active': {
            background: colors.button.primary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.primary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-pink']
        },
        '&$disabled': {
            background: colors.button.primary.bg.disabled,
            color: colors.text.white.secondary,
            cursor: 'not-allowed'
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    secondary: {
        background: colors.button.secondary.bg.default,
        color: colors.text.indigo.primary,
        '&:hover': {
            background: colors.button.secondary.bg.hover
        },
        '&:active': {
            background: colors.button.secondary.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '&$disabled': {
            background: colors.button.secondary.bg.disabled,
            color: colors.text.indigo.secondary
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiary: {
        background: 'transparent',
        color: colors.text.indigo.primary,
        outline: `2px solid ${colors.button.tertiary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.indigo.secondary,
            outline: `2px solid ${colors.button.tertiary.border.disabled}`
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiaryWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.tertiary.border.inverse.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.inverse.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.inverse.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    ghostPink: {
        background: 'transparent',
        color: sec,
        border: 'none',
        paddingLeft: [0, '!important'],
        paddingRight: [0, '!important'],
        '&:hover, &:active': {
            color: secDark,
            '& span span': {
                background: props => renderIcon(props, [secDark])
            },
        },
        '& span span': {
            background: props => renderIcon(props, [sec])
        },
        '&$disabled': {
            background: `transparent !important`,
            color: `${secLight} !important`,
            border: `none !important`,
            '& span span': {
                background: props => renderIcon(props, [grey500])
            }
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([sec]))
        }
    },
    ghostGrey: {
        background: 'transparent',
        color: colors.text.corp.secondary,
        '&:hover': {
            background: colors.button.ghost.bg.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-corp']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.default.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.corp.disabled
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.default.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.default.default]))
        }
    },
    ghostWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        '&:hover': {
            background: colors.button.ghost.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.inverse.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    loading: {
        pointerEvents: 'none',
        '& $cont': {
            visibility: 'hidden'
        }
    },
    disabled: {
        cursor: 'default',
        pointerEvents: 'none',
    },
    // Sizes
    md: {
        font: `${md.fontWeight} ${md.fontSize}/${md.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-5']],
        '&$iconOnly': {
            padding: [spacing['size-2'], spacing['size-2']]
        }
    },
    lg: {
        font: `${lg.fontWeight} ${lg.fontSize}/${lg.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-6']],
        '&$iconOnly': {
            padding: [spacing['size-3'], spacing['size-3']]
        }
    },
    // Block
    block: {
        display: 'block',
        width: '100%'
    },
    // Icon
    icon: {
        marginRight: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconRight: {
        width: 16,
        height: 16,
        marginLeft: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconOnly: {
        padding: [spacing['size-2'], spacing['size-2']],
        '& span span, & svg': {
            marginRight: 0
        }
    },
    loadIcon: {
        width: 24,
        height: 24,
        fill: colors.icon.inverse.default
    },
    round: {
        borderRadius: '50%'
    }
};
