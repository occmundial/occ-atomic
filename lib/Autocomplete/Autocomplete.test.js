import React from 'react';
import { shallow, mount } from 'enzyme';
import Autocomplete from './Autocomplete';
import styles from './styles';
import AutocompleteJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });
const classes = Object.keys(styles).reduce(reduceClasses, {});

const items = [
  { text: 'Administrador', textRight: '(222)', id: 1 },
  { text: 'Administrador Sr', textRight: '(134)', id: 2 },
  { text: 'Administrador Jr', textRight: '(52)', id: 3 },
  { text: 'Administrar', textRight: '(6)', id: 4 }
];

describe('Autocomplete', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(
      <Autocomplete droplistProps={{ items }} classes={classes} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the input and droplist', () => {
    const wrapper = mount(
      <Autocomplete
        textfieldProps={{ valueProp: 'a', className: 'input' }}
        droplistProps={{ items, className: 'droplist' }}
        classes={classes}
      />
    );
    wrapper.setState({ focus: true });
    expect(wrapper.find('.input').at(0).length).toBe(1);
    expect(wrapper.find('.droplist').at(0).length).toBe(1);
  });

  it("hides the droplist when there's no value or the input has no focus", () => {
    const wrapper = mount(
      <Autocomplete
        textfieldProps={{ valueProp: 'a', className: 'input' }}
        droplistProps={{ items, className: 'droplist' }}
        classes={classes}
      />
    );
    wrapper.setState({ focus: true });
    expect(wrapper.find('.droplist').at(0).length).toBe(1);
    wrapper.setState({ focus: false });
    expect(wrapper.find('.droplist').at(0).length).toBe(0);
    wrapper.setState({ focus: true, value: '' });
    expect(wrapper.find('.droplist').at(0).length).toBe(0);
  });

  it('filters the items of the droplist using the input value', () => {
    const wrapper = mount(
      <Autocomplete
        textfieldProps={{ className: 'input' }}
        droplistProps={{ items, className: 'droplist' }}
        classes={classes}
      />
    );
    wrapper.setState({ focus: true, value: 'a' });
    expect(wrapper.find('.droplist').at(0).text()).toBe(
      'Administrador(222)Administrador Sr(134)Administrador Jr(52)Administrar(6)'
    );
    wrapper.setState({ value: 'administrador' });
    expect(wrapper.find('.droplist').at(0).text()).toBe(
      'Administrador(222)Administrador Sr(134)Administrador Jr(52)'
    );
  });

  it('returns the input value', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <Autocomplete
        onChange={onChange}
        textfieldProps={{ className: 'input' }}
        droplistProps={{ items, className: 'droplist' }}
        classes={classes}
      />
    );
    wrapper.find('input').simulate('change', { target: { value: 'a' } });
    expect(onChange.mock.calls[0][0]).toBe('a');
    wrapper.find('input').simulate('change', { target: { value: 'ad' } });
    expect(onChange.mock.calls[1][0]).toBe('ad');
  });

  it('returns the object of the selected item', () => {
    const onMouseDown = jest.fn();
    const wrapper = mount(
      <Autocomplete
        onMouseDown={onMouseDown}
        textfieldProps={{ valueProp: 'a', className: 'input' }}
        droplistProps={{
          items,
          className: 'droplist',
          testId: 'test-component'
        }}
        classes={classes}
      />
    );
    wrapper.setState({ focus: true });
    wrapper.find({ "data-testid": 'test-component__droplist-item-0'}).simulate('mousedown');
    expect(onMouseDown.mock.calls[0][0]).toEqual({
      id: 1,
      text: 'Administrador',
      textRight: '(222)'
    });
    wrapper.setState({ focus: true, value: 'a' });
    wrapper.find({ "data-testid": 'test-component__droplist-item-1'}).simulate('mousedown');
    expect(onMouseDown.mock.calls[1][0]).toEqual({
      id: 2,
      text: 'Administrador Sr',
      textRight: '(134)'
    });
  });

  it('calls the onChange function', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <Autocomplete
        onChange={onChange}
        textfieldProps={{ valueProp: 'a', className: 'input' }}
        droplistProps={{ items, className: 'droplist' }}
        classes={classes}
      />
    );
    wrapper.find('input').simulate('change', { target: { value: 'a' } });
    expect(onChange.mock.calls.length).toBe(1);
  });

  it('calls the onMouseDown function', () => {
    const onMouseDown = jest.fn();
    const wrapper = mount(
      <Autocomplete
        onMouseDown={onMouseDown}
        textfieldProps={{ valueProp: 'a', className: 'input' }}
        droplistProps={{ items, className: 'droplist', testId: "test-component" }}
        classes={classes}
      />
    );
    wrapper.setState({ focus: true });
    wrapper.find({ "data-testid": 'test-component__droplist-item-0'}).simulate('mousedown');
    expect(onMouseDown.mock.calls.length).toBe(1);
  });

  it('calls the onFocus function', () => {
    const onFocus = jest.fn();
    const wrapper = mount(
      <Autocomplete
        onFocus={onFocus}
        textfieldProps={{ valueProp: 'a', className: 'input' }}
        droplistProps={{ items, className: 'droplist' }}
        classes={classes}
      />
    );
    wrapper.find('input').simulate('focus');
    expect(onFocus.mock.calls.length).toBe(1);
  });

  it('calls the onBlur function', () => {
    const onBlur = jest.fn();
    const wrapper = mount(
      <Autocomplete
        onBlur={onBlur}
        textfieldProps={{ valueProp: 'a', className: 'input' }}
        droplistProps={{ items, className: 'droplist' }}
        classes={classes}
      />
    );
    wrapper.find('input').simulate('blur');
    expect(onBlur.mock.calls.length).toBe(1);
  });
});

describe('AutocompleteJSS', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(
      <AutocompleteJSS droplistProps={{ items }} items={items} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Autocomplete styles', () => {
  it('matches the snapshot', () => {
    expect(styles).toMatchSnapshot();
  });
});
