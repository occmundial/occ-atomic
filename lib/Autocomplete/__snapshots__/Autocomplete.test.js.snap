// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Autocomplete
    classes={
      Object {
        "autoComplete": "autoComplete",
        "droplist": "droplist",
      }
    }
    disableAutoComplete={false}
    droplistProps={
      Object {
        "items": Array [
          Object {
            "id": 1,
            "text": "Administrador",
            "textRight": "(222)",
          },
          Object {
            "id": 2,
            "text": "Administrador Sr",
            "textRight": "(134)",
          },
          Object {
            "id": 3,
            "text": "Administrador Jr",
            "textRight": "(52)",
          },
          Object {
            "id": 4,
            "text": "Administrar",
            "textRight": "(6)",
          },
        ],
      }
    }
    showInitialData={false}
    textfieldProps={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Jss(TextField)
          autoFocus={false}
          clear={false}
          counter={false}
          disabled={false}
          hjWhitelist={true}
          onBlur={[Function]}
          onChange={[Function]}
          onClear={[Function]}
          onFocus={[Function]}
          onKeyUp={[Function]}
          onRef={[Function]}
          options={Array []}
          selectOnFocus={false}
          type="text"
          valueProp=""
        />,
        "",
      ],
      "className": "autoComplete",
      "id": undefined,
      "style": undefined,
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": false,
          "clear": false,
          "counter": false,
          "disabled": false,
          "hjWhitelist": true,
          "onBlur": [Function],
          "onChange": [Function],
          "onClear": [Function],
          "onFocus": [Function],
          "onKeyUp": [Function],
          "onRef": [Function],
          "options": Array [],
          "selectOnFocus": false,
          "type": "text",
          "valueProp": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Jss(TextField)
            autoFocus={false}
            clear={false}
            counter={false}
            disabled={false}
            hjWhitelist={true}
            onBlur={[Function]}
            onChange={[Function]}
            onClear={[Function]}
            onFocus={[Function]}
            onKeyUp={[Function]}
            onRef={[Function]}
            options={Array []}
            selectOnFocus={false}
            type="text"
            valueProp=""
          />,
          "",
        ],
        "className": "autoComplete",
        "id": undefined,
        "style": undefined,
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": false,
            "clear": false,
            "counter": false,
            "disabled": false,
            "hjWhitelist": true,
            "onBlur": [Function],
            "onChange": [Function],
            "onClear": [Function],
            "onFocus": [Function],
            "onKeyUp": [Function],
            "onRef": [Function],
            "options": Array [],
            "selectOnFocus": false,
            "type": "text",
            "valueProp": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Autocomplete styles matches the snapshot 1`] = `
Object {
  "autoComplete": Object {
    "position": "relative",
  },
  "droplist": Object {
    "left": 0,
    "position": "absolute",
    "top": [Function],
    "width": "100%",
    "zIndex": 2,
  },
}
`;

exports[`AutocompleteJSS matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Autocomplete)
    disableAutoComplete={false}
    droplistProps={
      Object {
        "items": Array [
          Object {
            "id": 1,
            "text": "Administrador",
            "textRight": "(222)",
          },
          Object {
            "id": 2,
            "text": "Administrador Sr",
            "textRight": "(134)",
          },
          Object {
            "id": 3,
            "text": "Administrador Jr",
            "textRight": "(52)",
          },
          Object {
            "id": 4,
            "text": "Administrar",
            "textRight": "(6)",
          },
        ],
      }
    }
    items={
      Array [
        Object {
          "id": 1,
          "text": "Administrador",
          "textRight": "(222)",
        },
        Object {
          "id": 2,
          "text": "Administrador Sr",
          "textRight": "(134)",
        },
        Object {
          "id": 3,
          "text": "Administrador Jr",
          "textRight": "(52)",
        },
        Object {
          "id": 4,
          "text": "Administrar",
          "textRight": "(6)",
        },
      ]
    }
    showInitialData={false}
    textfieldProps={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "autoComplete": "Autocomplete-autoComplete-0-1-77 Autocomplete-autoComplete-0-1-74",
        "droplist": "Autocomplete-droplist-0-1-76 Autocomplete-droplist-0-1-75",
      },
      "disableAutoComplete": false,
      "droplistProps": Object {
        "items": Array [
          Object {
            "id": 1,
            "text": "Administrador",
            "textRight": "(222)",
          },
          Object {
            "id": 2,
            "text": "Administrador Sr",
            "textRight": "(134)",
          },
          Object {
            "id": 3,
            "text": "Administrador Jr",
            "textRight": "(52)",
          },
          Object {
            "id": 4,
            "text": "Administrar",
            "textRight": "(6)",
          },
        ],
      },
      "items": Array [
        Object {
          "id": 1,
          "text": "Administrador",
          "textRight": "(222)",
        },
        Object {
          "id": 2,
          "text": "Administrador Sr",
          "textRight": "(134)",
        },
        Object {
          "id": 3,
          "text": "Administrador Jr",
          "textRight": "(52)",
        },
        Object {
          "id": 4,
          "text": "Administrar",
          "textRight": "(6)",
        },
      ],
      "showInitialData": false,
      "textfieldProps": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "autoComplete": "Autocomplete-autoComplete-0-1-77 Autocomplete-autoComplete-0-1-74",
          "droplist": "Autocomplete-droplist-0-1-76 Autocomplete-droplist-0-1-75",
        },
        "disableAutoComplete": false,
        "droplistProps": Object {
          "items": Array [
            Object {
              "id": 1,
              "text": "Administrador",
              "textRight": "(222)",
            },
            Object {
              "id": 2,
              "text": "Administrador Sr",
              "textRight": "(134)",
            },
            Object {
              "id": 3,
              "text": "Administrador Jr",
              "textRight": "(52)",
            },
            Object {
              "id": 4,
              "text": "Administrar",
              "textRight": "(6)",
            },
          ],
        },
        "items": Array [
          Object {
            "id": 1,
            "text": "Administrador",
            "textRight": "(222)",
          },
          Object {
            "id": 2,
            "text": "Administrador Sr",
            "textRight": "(134)",
          },
          Object {
            "id": 3,
            "text": "Administrador Jr",
            "textRight": "(52)",
          },
          Object {
            "id": 4,
            "text": "Administrar",
            "textRight": "(6)",
          },
        ],
        "showInitialData": false,
        "textfieldProps": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
